### PR TITLE
cmd/cored: read TLS cert and key from files

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

**Comment by kr** [comment]
PTAL

File: cmd/cored/main.go [hunk 1]
```diff
@@ -255,6 +246,31 @@ func runServer() {
 	select {}
 }
 
+// maybeUseTLS loads the TLS cert and key (if so configured)
+// and wraps ln in a TLS listener.
+func maybeUseTLS(ln net.Listener) (net.Listener, bool, error) {
```
Comment (tessr) [inline cmd/cored/main.go]: nit: either name the return values or explain the bool in the comment

**Comment by tessr** [comment]
👀

File: cmd/cored/main.go [hunk 2]
```diff
@@ -255,6 +246,31 @@ func runServer() {
...
+	keyFile := filepath.Join(*dataDir, "tls.key")
+
+	_, certErr := os.Lstat(certFile)
+	_, keyErr := os.Lstat(keyFile)
+	if os.IsNotExist(certErr) && os.IsNotExist(keyErr) {
+		return ln, false, nil // files don't exist; don't want TLS
+	}
+
+	config := &tls.Config{
+		NextProtos: []string{"http/1.1", "h2"},
```
Comment (tessr) [inline cmd/cored/main.go]: maybe explain this too

File: cmd/cored/main.go [hunk 3]
```diff
@@ -255,6 +246,31 @@ func runServer() {
 	select {}
 }
 
+// maybeUseTLS loads the TLS cert and key (if so configured)
+// and wraps ln in a TLS listener.
+func maybeUseTLS(ln net.Listener) (net.Listener, bool, error) {
+	certFile := filepath.Join(*dataDir, "tls.crt")
```
Comment (boymanjor) [inline cmd/cored/main.go]: Do you think we should create a subdir for the certs?

**Comment by kr** [comment]
I'm going to update this to also support the env vars for now, and we'll call them deprecated, and not remove them until the next release at the earliest.

File: cmd/cored/main.go (hunk 3 above)
Comment (kr) [inline cmd/cored/main.go]: That wouldn't be my first choice. I would rather they be more visible.

File: cmd/cored/main.go (hunk 1 above)
Comment (kr) [inline cmd/cored/main.go]: Oh actually I'm going to remove that return param.

**Comment by kr** [comment]
Updated to address feedback. PTAL.

**Comment by boymanjor** [comment]
LGTM

**Pull request merged**